src/net.cpp

```diff
@@ -1877,8 +1887,10 @@ void CConnman::ThreadMessageHandler()
                 pnode->Release();
         }
 
-        if (fSleep)
-            messageHandlerCondition.timed_wait(lock, boost::posix_time::microsec_clock::universal_time() + boost::posix_time::milliseconds(100));
+        if (fSleep) {
+            std::unique_lock<std::mutex> lock(mutexMsgProc);
+            condMsgProc.wait_until(lock, std::chrono::steady_clock::now() + std::chrono::milliseconds(100));
```

Comment:
It seems that `mutexMsgProc` is only used for locking `flagInterruptMsgProc` and waiting, making it very similar to `CThreadInterrupt`. If you'd add a `wake()` method to that class, which breaks from sleeps, but doesn't set the interrupt point, wouldn't you be able to replace the explicit mutex/flag/condvar with a CThreadInterrupt>